src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -10,21 +11,26 @@ namespace AutoRest.CSharp.Mgmt.Output
 {
     internal class Resource : TypeProvider
     {
-        public Resource(OperationGroup operationGroup, BuildContext context)
+        private BuildContext<MgmtOutputLibrary> _context;
+
+        public Resource(OperationGroup operationGroup, BuildContext<MgmtOutputLibrary> context)
             : base(context)
         {
-            OperationGroup = operationGroup;
             DefaultName = operationGroup.Resource(context.Configuration.MgmtConfiguration);
             Description = BuilderHelpers.EscapeXmlDescription(
                 $"A Class representing a {DefaultName} along with the instance operations that can be performed on it.");
+            _context = context;
+            OperationGroup = operationGroup;
         }
 
         protected override string DefaultName { get; }
 
         protected override string DefaultAccessibility => "public";
 
-        internal OperationGroup OperationGroup { get; }
-
         public string Description { get; }
+
+        internal OperationGroup OperationGroup { get; private set; }
```

Comment:
Why do you need private setter here? I think it can be changed to 
>internal OperationGroup OperationGroup { get; }

Reply:
Actually you don't need to add OperationGroup in this class. It is already added, check [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Output/Resource.cs#L22). You just need to rebase.